basics_of_sst-core/tutorial/Example00.md

```diff
@@ -0,0 +1,568 @@
+
+Next the logger is created.  For more information on how the logger is configured and used, consult the SST documentation.
+
+```
+    // Create the logger.
+    //
+    logger_ = SST::Output("Time=@t; File=@f; Func=@p; Line=@l; Thread=@I -- ", debug, 0x01, SST::Output::STDOUT);
+    logger_.verbose(CALL_INFO, TRACE, 0x00, "Entering constructor for component id %lu\n", componentId_);
+
+```
+
+After the logger, the component clock is configured.  This simply consists of registering the clock and specifying the clock event handler.  The clock event handler is called each clock period and is discussed below.
+
+```
+    // Configure the component clock.
+    //
+    logger_.verbose(CALL_INFO, DEBUG, 0x00, "Clock rate is: %s\n", clock.c_str());
+    registerClock(clock,
+        new SST::Clock::Handler<ExampleComponent>(this, &ExampleComponent::clockTick));
+    logger_.verbose(CALL_INFO, INFO,  0x00, "Successfully initialized clock.\n");
+```
+
+Finally the component is registered with the simulation.
+
+```
+    // Register this component with the simulation.
+    //
+    registerAsPrimaryComponent();
+    logger_.verbose(CALL_INFO, DEBUG, 0x00, "Component registered as primary component.\n");
+    primaryComponentDoNotEndSim();
+    logger_.verbose(CALL_INFO, DEBUG, 0x00, "Simulation notified it should not end.\n");
+    logger_.verbose(CALL_INFO, TRACE, 0x00, "Leaving constructor for component id %lu\n", componentId_);
+```
+
+Registering the component as a primary component allows this component to specify when it is ok to end the simulation through the use of the `primaryComponentDoNotEndSim()` and `primaryComponentOKToEndSim()` calls.  The simulation will not end naturally until all primary components specify it is ok to end the simulation.
+
+### Understanding setup() and finish()
+
+As described previously, `setup()` and `finish()` are where additional component initialization and cleanup take place.  They are called prior to the simulation starting and after the simulation ends, respectively.  In this case, there is no initialization or cleanup needed so they simply contain logging statements to indicate they were called.
+
+```
+// Called after all components have been constructed and initialization
+// has completed, but before simulation time has begin.
+//
+// This is where you should do any other initialization that needs done
+// but could be accomplished in the constructure.
+//
+void ExampleComponent::setup(void)
+{
+    logger_.verbose(CALL_INFO, TRACE, 0x00, "Entering setup for component id %lu\n", componentId_);
+    logger_.verbose(CALL_INFO, TRACE, 0x00, "Leaving setup for component id %lu\n", componentId_);
+}
+
+
+// Called after the simulation is complete but before the objects are
+// destroyed.  This is a good place to print out statistics.
```

Comment:
Before SST contained built-in statistics support, this was the place to do statistics. Now however, it's recommended to use the Statistics API for statistics and use finish() for non-statistic output or to update statistics (through the Statistics API) if needed before they are printed by SST.